src/main/flight/pid.c

```diff
@@ -842,6 +840,83 @@ void FAST_CODE applySmartFeedforward(int axis)
 }
 #endif // USE_SMART_FEEDFORWARD
 
+#if defined(USE_ABSOLUTE_CONTROL)
+static void applyAbsoluteControl(const int axis, const float gyroRate,
+    const float setpointLpf, const float setpointHpf,
+    float *currentPidSetpoint, float *itermErrorRate)
+{
+    if (acGain > 0) {
+        float acCorrection = 0;
+        float acErrorRate;
+
+        const float gmaxac = setpointLpf + 2 * setpointHpf;
+        const float gminac = setpointLpf - 2 * setpointHpf;
+        if (gyroRate >= gminac && gyroRate <= gmaxac) {
+            float acErrorRate1 = gmaxac - gyroRate;
+            float acErrorRate2 = gminac - gyroRate;
+            if (acErrorRate1 * axisError[axis] < 0) {
+                acErrorRate = acErrorRate1;
+            } else {
+                acErrorRate = acErrorRate2;
+            }
+            if (fabsf(acErrorRate * dT) > fabsf(axisError[axis]) ) {
+                acErrorRate = -axisError[axis] / dT;
```

Comment:
Should be changed to use `samplingFrequency` which we had cached somewhere

Reply:
Yes.....multiply can than be used, but still I recall the original creator to go through the code. 
When playing with unittests I was even not able to hit some of the if scenarios in the code.

Is there anybody who understands absolute control fully?